x-pack/plugins/endpoint/public/applications/endpoint/store/policy_list/test_mock_utils.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpStart } from 'kibana/public';
+import { Dispatch } from 'redux';
+import { INGEST_API_DATASOURCES } from './services/ingest';
+import { EndpointDocGenerator } from '../../../../../common/generate_data';
+import { AppAction, GetDatasourcesResponse, GlobalState, MiddlewareFactory } from '../../types';
+
+const generator = new EndpointDocGenerator('policy-list');
+
+/**
+ * It sets the mock implementation on the necessary http methods to support the policy list view
+ * @param mockedHttpService
+ * @param responseItems
+ */
+export const setPolicyListApiMockImplementation = (
+  mockedHttpService: jest.Mocked<HttpStart>,
+  responseItems: GetDatasourcesResponse['items'] = [generator.generatePolicyDatasource()]
+): void => {
+  mockedHttpService.get.mockImplementation((...args) => {
+    const [path] = args;
+    if (typeof path === 'string') {
+      if (path === INGEST_API_DATASOURCES) {
+        return Promise.resolve<GetDatasourcesResponse>({
+          items: responseItems,
+          total: 10,
+          page: 1,
+          perPage: 10,
+          success: true,
+        });
+      }
+    }
+    return Promise.reject(new Error(`MOCK: unknown policy list api: ${path}`));
+  });
+};
+
+/**
+ * Utilities for testing Redux middleware
+ */
+export interface MiddlewareActionSpyHelper<S = GlobalState> {
+  /**
+   * Returns a promise that is fulfilled when the given action is dispatched or a timeout occurs.
+   * The use of this method instead of a `sleep()` type of delay should avoid test case instability
+   * especially when run in a CI environment.
+   *
+   * @param actionType
+   */
+  waitForAction: (actionType: Pick<AppAction, 'type'>['type']) => Promise<void>;
+  /**
+   * A property holding the information around the calls that were processed by the  internal
+   * `actionSpyMiddlware`. This property holds the information typically found in Jets's mocked
+   * function `mock` property - [see here for more information](https://jestjs.io/docs/en/mock-functions#mock-property)
+   *
+   * **Note**: this property will only be set **after* the `actionSpyMiddlware` has been
+   * initialized (ex. via `createStore()`. Attempting to reference this property before that time
+   * will throw an error.
+   * Also - do not hold on to references to this property value if `jest.clearAllMocks()` or
+   * `jest.resetAllMocks()` is called between usages of the value.
+   */
+  dispatchSpy: jest.Mock<Dispatch<AppAction>>['mock'];
```

Comment:
if this isn't always set, maybe make the property `dispatchSpy?` instead? that way we'll check for validity before using it (and hopefully avoid an error.)

Reply:
yeah, thats another alternative. But would that be a worst developer experience? It would imply that we would need to either check the property every time we try to use it (`utils!.dispatchSpy`) or in destructing it, like `dispatchSpy = utils!.dispatchUtils` which could lead to an error if done prior to store creation.